idioms/str-vs-string.md

```diff
@@ -0,0 +1,89 @@
+# Using &str instead of &String
+
+## Description
+
+It is usually recommended to prefer a `&str` type over a `&String` type for arguments.  In most cases `&str` has more flexibility.  A `&String` will coerce to a `&str` type when required -- however, the opposite is not true.  
```

Comment:
And also again, it is possible (though I'm not aware of anyone actually doing this) for alternate `String` implementations to also implement `Deref<Target=str>` and thus taking `&str` accommodates this use case too.